src/target/riscv/riscv_reg_impl.h

```diff
@@ -0,0 +1,211 @@
+#ifndef OPENOCD_TARGET_RISCV_RISCV_REG_IMPL_H
```

Comment:
1) How about calling this `riscv_reg_helpers.h`, `riscv_reg_utils.h` or similar?

2) I would prefer to avoid the `static inline` function definitions in the header file. Instead, I'd recommend to keep the .h/.c structure for this module, too, so that all the modules look the same way and no irregularities are introduced.

3) If you like the naming convention for functions (mentioned in a previous comment), I'd recommend to follow it also here: `riscv_reg_utils_*()`.

Reply:
1. I'd like to leave the `_impl` suffix.
IMHO, `_utils` is generally used for useful functionality used throughout the codebase.
Here `_impl` denotes that these headers should only be included in implementations of register cache functionality, not for general use.

2. There are currently 3 `static inline` functions here. I'd like to leave it as-is.
      1. `riscv_reg_impl_is_initialized()` -- it's quite short and I'd like for it to be immediately clear what it means for the register to be "initialized".
      2. `riscv_reg_impl_init_vector_reg_type()` -- as mentioned in a `TODO` there, I'd like to move that function. So, leaving it in the header makes it more of an eyesore and makes me want to address the `TODO` quicker :). If you insist, I'll move it.
      3. `riscv_reg_impl_gdb_regno_cacheable()` -- this one will get removed by future changes. If you insist, I'll move it, but I'd like to leave it there for a reason similar to (2.ii).
3. Addressed, considering (1).

Reply:
1) I've now understood why the `_impl` suffix makes more sense in this case, thanks.

2) Understood, let's leave it as is.

3) Thanks.